node/network/availability-recovery/src/lib.rs

```diff
@@ -150,96 +147,111 @@ struct Interaction {
 }
 
 impl Interaction {
-	async fn run(mut self) {
-		loop {
-			// If it's empty and requesting_chunks is empty,
-			// break and issue a FromInteraction::Concluded(RecoveryError::Unavailable)
-			if self.requesting_chunks.is_empty() && self.shuffling.is_empty() {
-				if self.to_state.send(FromInteraction::Concluded(
+	async fn maybe_done(&mut self) -> error::Result<bool> {
```

Comment:
style: this function mixes the computation of being done with the effect that's meant to occur upon being done. It would be better to write this as a pure function and dispatch the message at the point it's called.

It also doesn't quite do what the name suggests. maybe rename to `is_impossible` or `is_unavailable`?

Reply:
still stands